packages/destination-actions/src/destinations/blackbaud-raisers-edge-nxt/utils/index.ts

```diff
@@ -17,6 +29,178 @@ export const dateStringToFuzzyDate = (dateString: string | number) => {
+}
+
+export const buildGiftDataFromPayload = (constituentId: string, payload: CreateGiftPayload) => {
+  // data for gift call
+  const giftData: Partial<Gift> = {
+    amount: payload.amount,
+    constituent_id: constituentId,
+    date: payload.date,
+    gift_status: payload.gift_status,
+    is_anonymous: payload.is_anonymous,
+    // hardcode is_manual
+    is_manual: true,
+    lookup_id: payload.lookup_id,
+    post_date: payload.post_date,
+    post_status: payload.post_status,
+    subtype: payload.subtype,
+    type: payload.type
+  }
+  Object.keys(giftData).forEach((key) => {
+    if (!giftData[key as keyof Gift]) {
+      delete giftData[key as keyof Gift]
+    }
+  })
+
+  // default date
+  giftData.date = giftData.date || new Date().toISOString()
+
+  // create acknowledgements array
+  if (payload.acknowledgement) {
+    const acknowledgementData: GiftAcknowledgement = {
+      status: payload.acknowledgement.status || 'NEEDSACKNOWLEDGEMENT'
+    }
+    if (
+      acknowledgementData.status !== 'NEEDSACKNOWLEDGEMENT' &&
+      acknowledgementData.status !== 'DONOTACKNOWLEDGE' &&
+      payload.acknowledgement.date
+    ) {
+      acknowledgementData.date = payload.acknowledgement.date
+    }
+    giftData.acknowledgements = [acknowledgementData]
+  }
+
+  // create gift splits array
+  giftData.gift_splits = [
+    {
+      amount: payload.amount,
+      fund_id: payload.fund_id
+    }
+  ]
+
+  // create payments array
+  giftData.payments = [
+    {
+      payment_method: payload.payment_method
+    }
+  ]
+
+  // fields for check gifts
+  if (giftData.payments[0].payment_method === 'PersonalCheck') {
+    giftData.payments[0].check_number = payload.check_number
+    if (payload.check_date) {
+      const checkDateFuzzyDate = dateStringToFuzzyDate(payload.check_date)
+      if (checkDateFuzzyDate) {
+        giftData.payments[0].check_date = checkDateFuzzyDate
+      }
+    }
+  }
+
+  // default post date
+  if ((giftData.post_status === 'NotPosted' || giftData.post_status === 'Posted') && !giftData.post_date) {
+    giftData.post_date = payload.date
+  }
+
+  // create receipts array
+  if (payload.receipt) {
+    const receiptData: GiftReceipt = {
+      status: payload.receipt.status || 'NEEDSRECEIPT'
+    }
+    if (receiptData.status === 'RECEIPTED' && payload.receipt.date) {
+      receiptData.date = payload.receipt.date
+    }
+    giftData.receipts = [receiptData]
+  }
+
+  // fields for recurring gifts
+  if (giftData.type === 'RecurringGift') {
+    if (payload.recurring_gift_schedule) {
+      giftData.recurring_gift_schedule = {
+        end_date: payload.recurring_gift_schedule.end_date,
+        frequency: payload.recurring_gift_schedule.frequency || '',
+        start_date: payload.recurring_gift_schedule.start_date || ''
+      }
+    }
+  } else if (giftData.type === 'RecurringGiftPayment' && payload.linked_gifts) {
+    giftData.linked_gifts = payload.linked_gifts.split(',')
```

Comment:
Just want to clarify if we expect the customer to pass in a comma separated string in the original segment call? we have something similar for fundraisers in the Create Action call

Reply:
Yep. Segment events don't allow an `array` data type, so that's the only way I could come up with.